### PR TITLE
mgr/orchestrator: Better align Drive Groups with DeepSea

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/pybind/mgr/orchestrator.py [hunk 1]
```diff
@@ -635,6 +665,12 @@ def __init__(self, host_pattern, data_devices=None, db_devices=None, wal_devices
         #: A :class:`orchestrator.DeviceSelection`
         self.journal_devices = journal_devices
 
+        # TODO: is this actually implemented in DeepSea?
+        self.wal_size = wal_size
```
Comment (jschmid1) [inline src/pybind/mgr/orchestrator.py]: Yes, it's called block_wal/db_size, to match the naming `ceph-volume` uses.

I'm open for discussion if we should just accept both versions.

**Comment by jschmid1** [comment]
This looks like we're moving closer the the implementation in DeepSea.

What I still don't really like is the use of a `path` spec. Using fixed paths opens all kinds of problems for day-2 operations (if the drivegroup is not used ephemerally). It's entirely possible to describe all kinds of combinations of disks by just using the mentioned specifications.

File: src/pybind/mgr/orchestrator.py (hunk 1 above)
Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator.py]: Hm, would you be ok with just using the names of ceph-volume here? this way we're not inventing a new name for the same thing.

**Comment by sebastian-philipp** [comment]
> This looks like we're moving closer the the implementation in DeepSea.
> 
> What I still don't really like is the use of a `path` spec. Using fixed paths opens all kinds of problems for day-2 operations (if the drivegroup is not used ephemerally). It's entirely possible to describe all kinds of combinations of disks by just using the mentioned specifications.

Problem is that there are existing (non enterprise) use cases for path. I'm going to add a warning that this should not be used in production environments.

**Comment by jschmid1** [comment]
> Problem is that there are existing (non enterprise) use cases for path. I'm going to add a warning that this should not be used in production environments.

could you elaborate on this? I reckon it's not necessary at all to use `path`

**Comment by sebastian-philipp** [comment]
> > Problem is that there are existing (non enterprise) use cases for path. I'm going to add a warning that this should not be used in production environments.
> 
> could you elaborate on this? I reckon it's not necessary at all to use `path`

[Rook's CephCluster CRD](https://github.com/rook/rook/blob/master/Documentation/ceph-cluster-crd.md#storage-selection-settings) has a directory attribute that allows using directories. Having this field in the structure doesn't necessarily mean that all orchestrators need to implement this.

**Comment by sebastian-philipp** [comment]
rebased to python-common

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-09-07_14:44:52-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-09-08_13:41:17-rados-wip-kefu-testing-2019-09-07-0159-distro-basic-smithi/

@jschmid1 ping?

**Comment by jschmid1** [comment]
looks fine @tchaikov @sebastian-philipp

**Comment by sebastian-philipp** [comment]
jenkins retest this please

File: src/python-common/ceph/deployment/drive_group.py [hunk 2]
```diff
@@ -16,20 +16,27 @@ class DeviceSelection(object):
...
 
         #: A wildcard string. e.g: "SDD*"
         self.id_model = id_model
 
+        #: Match on the VENDOR property of the drive
+        self.vendor = vendor
+
         #: Size specification of format LOW:HIGH.
         #: Can also take the the form :HIGH, LOW:
         #: or an exact value (as ceph-volume inventory reports)
```
Comment (bk201) [inline src/python-common/ceph/deployment/drive_group.py]: Can we also make comment for `size` clearer? For example, to support the same flavor as the drive group in DeepSea. This not only affects how orchestrator backend implements related features but also affects how clients (like Dashboard) pass the value. Thanks!

**Comment by sebastian-philipp** [comment]
Replaced with #31021

**Pull request closed without merging**